codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/XmlMemberDeserVisitor.java

```diff
@@ -0,0 +1,370 @@
+package software.amazon.smithy.aws.go.codegen;
+
+import java.util.function.Predicate;
+import software.amazon.smithy.codegen.core.CodegenException;
+import software.amazon.smithy.codegen.core.Symbol;
+import software.amazon.smithy.go.codegen.CodegenUtils;
+import software.amazon.smithy.go.codegen.GoWriter;
+import software.amazon.smithy.go.codegen.SmithyGoDependency;
+import software.amazon.smithy.go.codegen.integration.ProtocolGenerator;
+import software.amazon.smithy.go.codegen.integration.ProtocolGenerator.GenerationContext;
+import software.amazon.smithy.model.shapes.BigDecimalShape;
+import software.amazon.smithy.model.shapes.BigIntegerShape;
+import software.amazon.smithy.model.shapes.BlobShape;
+import software.amazon.smithy.model.shapes.BooleanShape;
+import software.amazon.smithy.model.shapes.ByteShape;
+import software.amazon.smithy.model.shapes.CollectionShape;
+import software.amazon.smithy.model.shapes.DocumentShape;
+import software.amazon.smithy.model.shapes.DoubleShape;
+import software.amazon.smithy.model.shapes.FloatShape;
+import software.amazon.smithy.model.shapes.IntegerShape;
+import software.amazon.smithy.model.shapes.ListShape;
+import software.amazon.smithy.model.shapes.LongShape;
+import software.amazon.smithy.model.shapes.MapShape;
+import software.amazon.smithy.model.shapes.MemberShape;
+import software.amazon.smithy.model.shapes.OperationShape;
+import software.amazon.smithy.model.shapes.ResourceShape;
+import software.amazon.smithy.model.shapes.ServiceShape;
+import software.amazon.smithy.model.shapes.SetShape;
+import software.amazon.smithy.model.shapes.Shape;
+import software.amazon.smithy.model.shapes.ShapeVisitor;
+import software.amazon.smithy.model.shapes.ShortShape;
+import software.amazon.smithy.model.shapes.StringShape;
+import software.amazon.smithy.model.shapes.StructureShape;
+import software.amazon.smithy.model.shapes.TimestampShape;
+import software.amazon.smithy.model.shapes.UnionShape;
+import software.amazon.smithy.model.traits.EnumTrait;
+import software.amazon.smithy.model.traits.TimestampFormatTrait.Format;
+
+/**
+ * Visitor to generate member values for aggregate types deserialized from documents.
+ */
+public class XmlMemberDeserVisitor implements ShapeVisitor<Void> {
+    private final GenerationContext context;
+    private final String dataDest;
+    private final Format timestampFormat;
+    private final Predicate<Shape> filter;
+    private final Boolean isFlattened;
+
+
+    public XmlMemberDeserVisitor(GenerationContext context, String dataDest, Format timestampFormat, Predicate<Shape> filter, Boolean isFlattened) {
+        this.context = context;
+        this.dataDest = dataDest;
+        this.timestampFormat = timestampFormat;
+        this.filter = filter;
```

Comment:
The filter is to identify xml attributes. In case of xml attributes we do not need to decode a value enclosed between the start/end tags, but rather the attribute value enclosed within the start element. 

I have renamed filter to `xmlAttributeMemberFilter` and will add more documentation